src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -370,14 +372,42 @@ public Utf8String Substring(int index, int length)
             }
         }
 
-        // TODO: Should this be public or should focused on scenarios?
-        private int IndexOf(Utf8String value)
+        // TODO: Naive algorithm, reimplement faster
+        // TODO: Should this be public?
+        public int IndexOf(Utf8String value)
         {
-            throw new NotImplementedException();
+            if (value.Length == 0)
+            {
+                // TODO: Is this the right answer?
+                // TODO: Does this even make sense?
+                return 0;
+            }
+
+            if (Length == 0)
+            {
+                return StringNotFound;
+            }
+
+            Utf8String restOfTheString = this;
+            for (int i = 0; restOfTheString.Length <= Length; restOfTheString = Substring(++i))
```

Comment:
Take a look at code gen and see if it's as simple as incrementing an int, or if it's simple enough that it could be optimized. If the later, then you should chat with the compiler team[s] to see if they can optimize it.